### PR TITLE
Add ``minimal`` option to ``pylint-config``

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Write a good description on what the PR does.
- [x] Create a news fragment with `towncrier create <IssueNumber>.<type>` which will be
  included in the changelog. `<type>` can be one of: new_check, removed_check, extension,
  false_positive, false_negative, bugfix, other, internal. If necessary you can write
  details or offer examples on how the new change is supposed to work.
- [x] If you used multiple emails or multiple names when contributing, add your mails
      and preferred name in ``script/.contributors_aliases.json``

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |

## Description

Closes https://github.com/PyCQA/pylint/issues/7485

@Pierre-Sassoulas Shall we back port this so we can use it in `astroid`?

### CODE REVIEW

File: pylint/config/_pylint_config/generate_command.py [hunk 1]
```diff
@@ -22,10 +22,14 @@ def generate_interactive_config(linter: PyLinter) -> None:
     print("Starting interactive pylint configuration generation")
 
     format_type = utils.get_and_validate_format()
+    if format_type == "toml":
+        minimal = utils.get_minimal_setting()
+    else:
+        minimal = False
```
Comment (cclauss) [inline pylint/config/_pylint_config/generate_command.py:28]:
```suggestion
    minimal = format_type == "toml" and utils.get_minimal_setting()
```

File: tests/config/pylint_config/test_pylint_config_generate.py [hunk 2]
```diff
@@ -150,3 +159,34 @@ def test_writing_to_output_file(
...
+    # Set the answers needed for the input() calls
+    answers = iter(["no", "yes"])
+    monkeypatch.setattr("builtins.input", lambda x: next(answers))
+
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", message="NOTE:.*", category=UserWarning)
+        # Check not minimal has comments
+        Run(["generate", "--interactive"], exit=False)
+        captured = capsys.readouterr()
+        assert any(i.startswith("#") for i in captured.out.split("\n"))
```
[the remark this answers is not in the document]
Comment (cclauss) [reply tests/config/pylint_config/test_pylint_config_generate.py:186]: Also, is `captured.out.splitlines()` safer across platforms than `captured.out.split("\n")`?

[the remark this answers is not in the document]
Comment (DanielNoord) [reply tests/config/pylint_config/test_pylint_config_generate.py]: Changes to `splitlines`

File: tests/config/pylint_config/test_pylint_config_generate.py [hunk 3]
```diff
@@ -150,3 +159,34 @@ def test_writing_to_output_file(
...
+        warnings.filterwarnings("ignore", message="NOTE:.*", category=UserWarning)
+        # Check not minimal has comments
+        Run(["generate", "--interactive"], exit=False)
+        captured = capsys.readouterr()
+        assert any(line.startswith("#") for line in captured.out.splitlines())
+
+        # Check minimal doesn't have comments and no default values
+        Run(["--accept-no-return-doc=y", "generate", "--interactive"], exit=False)
+        captured = capsys.readouterr()
+        assert not any(i.startswith("#") for i in captured.out.split("\n"))
```
Comment (cclauss) [inline tests/config/pylint_config/test_pylint_config_generate.py:191]:
```suggestion
        assert not any(line.startswith("#") for line in captured.out.split lines())
```

**Review by cclauss — APPROVED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]
LGTM ! But I don't think we should backport : it's a new feature (and we can launch the dev version of pylint if we need it ourselves before 2.16.0).

**Pull request merged**